module_utils/network_lsr/argument_validator.py

```diff
@@ -2012,3 +2071,30 @@ def _ipv6_is_not_configured(connection):
                             "match.path is not supported by the running version of "
                             "NetworkManger.",
                         )
+        self.validate_route_tables(connection, idx)
+
+
+class RouteTableParsing:
+    @classmethod
+    def parse_route_tables_mapping(cls, file_content):
+        table_cache = {}
+        for table in file_content.split("\n"):
+            table_mapping = table.split()
+            # This is the original code about the route tables parsing in iproute2,
+            # https://git.kernel.org/pub/scm/network/iproute2/iproute2.git/tree/lib/rt_names.c?id=ade99e208c1843ed3b6eb9d138aa15a6a5eb5219#n38
+            if len(table_mapping) == 2 and table[0] != "#" and table[0] != "0":
+                table_cache[table_mapping[1]] = table_mapping[0]
+        return table_cache
+
+    @classmethod
+    def parse_route_tables_mapping_from_file(cls, filename):
+        with open(filename, "r") as f:
+            return cls.parse_route_tables_mapping(f.read())
+
+    @classmethod
+    def get_route_tables_mapping(cls):
+        if not hasattr(cls, "_cached_rt_tables"):
+            cls._cached_rt_tables = cls.parse_route_tables_mapping_from_file(
```

Comment:
`parse_route_tables_mapping_from_file()` can raise an exception (if the file doesn't exist). The caching `get_route_tables_mapping()` should never fail. If the files doesn't exist, return an empty mapping (but still a usable dictionary, so you don't have to handle the failure specially).

Reply:
I updated.